### PR TITLE
Re-write valueType sub-tests

### LINKED ISSUE #16284: Tests in ValueTypeTests skipped in Valhalla builds
Found by @a7ehuo in Valhalla nightly build: https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdkValhalla_j9_sanity.functional_x86-64_linux_valhalla_Nightly_testList_0/673

Only tests that have priority=1 in `ValueTypeTests.java` get executed. All other tests with priority>1 are skipped unexpectedly.

### LINKED ISSUE #16291: ValueTypes: java.lang.VerifyError in ValueTypeTests_1 testACMPTestOnFastSubstitutableValueTypesVer2
After removing `<class name="org.openj9.test.lworld.ValueTypeArrayTests"/>` from `testng.xml`, I'm able to run more Valhalla functional tests. The reason that it has to be removed in order to run more tests is tracked in https://github.com/eclipse-openj9/openj9/issues/16284. 

`testACMPTestOnFastSubstitutableValueTypesVer2`  in `ValueTypeTests_1` failed due to `java.lang.VerifyError: JVMVRFY021 thrown object not throwable; class=FastSubVT, method=withGenericarr(Ljava/lang/Object;)Ljava/lang/Object;, pc=2; Type Mismatch, argument 0 in signature FastSubVT.witharr:([Ljava/lang/Object;)QFastSubVT; does not match`

```
1:28:13  Eclipse OpenJ9 VM (build test-fix-field-offset-flattened-arrayelement-bb482f0ec7b, JRE 20 Linux amd64-64-Bit Compressed References 20221108_261 (JIT enabled, AOT enabled)
11:28:13  OpenJ9   - bb482f0ec7b
11:28:13  OMR      - 1c49c4a411f
11:28:13  JCL      - 44115f48aff based on jdk-20+19)
```

```
===============================================
ValueTypeTests_1 Start Time: Tue Nov  8 08:35:59 2022 Epoch Time (ms): 1667925359386
variation: -Xgcpolicy:optthruput -XX:ValueTypeFlatteningThreshold=99999 -XX:-EnableArrayFlattening
JVM_OPTIONS:  -Xgcpolicy:optthruput -XX:ValueTypeFlatteningThreshold=99999 -XX:-EnableArrayFlattening 
....
FAILED: testACMPTestOnFastSubstitutableValueTypesVer2
java.lang.VerifyError: JVMVRFY021 thrown object not throwable; class=FastSubVT, method=withGenericarr(Ljava/lang/Object;)Ljava/lang/Object;, pc=2; Type Mismatch, argument 0 in signature FastSubVT.witharr:([Ljava/lang/Object;)QFastSubVT; does not match
Exception Details:
  Location:
    FastSubVT.withGenericarr(Ljava/lang/Object;)Ljava/lang/Object; @2: JBinvokevirtual
  Reason:
    Type 'java/lang/Object' (current frame, stack[1]) is not assignable to '[Ljava/lang/Object;'
  Current Frame:
    bci: @2
    flags: { }
    locals: { 'FastSubVT', 'java/lang/Object' }
    stack: { 'FastSubVT', 'java/lang/Object' }
	at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
	at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1525)
	at java.base/java.lang.invoke.DirectMethodHandle.checkInitialized(DirectMethodHandle.java:387)
	at java.base/java.lang.invoke.DirectMethodHandle.ensureInitialized(DirectMethodHandle.java:375)
	at java.base/java.lang.invoke.DirectMethodHandle.internalMemberNameEnsureInit(DirectMethodHandle.java:340)
	at <unknown class>.<unknown method>(Unknown Source)
	at org.openj9.test.lworld.ValueTypeTests.testACMPTestOnFastSubstitutableValueTypesVer2(ValueTypeTests.java:882)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:580)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:716)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:988)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.testng.TestNG.privateMain(TestNG.java:1354)
	at org.testng.TestNG.main(TestNG.java:1323)
```

### PR DESCRIPTION
**Pull request by hangshao0**
1. Re-write testFlatLayoutsWithRecursiveLongs and
testACMPTestOnFastSubstitutableValueTypesVer2 using classes from Javac
instead of ASM generated classes.

2. Put the new classes in ValueTypeTestClasses, the class names of
ValueLong, DoubleLong, QuadLong and DoubleQuadLong now become 
org/openj9/test/lworld/ValueTypeTestClasses$ValueTypeLong,
org/openj9/test/lworld/ValueTypeTestClasses$ValueTypeDoubleLong,
org/openj9/test/lworld/ValueTypeTestClasses$ValueTypeQuadLong,
org/openj9/test/lworld/ValueTypeTestClasses$ValueTypeDoubleQuadLong.

3. Update the DDR tests to check for the new class names.

4. Split ValueTypeArrayTests from ValueTypeTests.

Closes #16291 #16292
Closes #16284

Signed-off-by: Hang Shao <hangshao@ca.ibm.com>

### CODE REVIEW

**Comment by hangshao0** [comment]
@llxia @hzongaro Can you review the change of commit `Split ValueTypeArrayTests from ValueTypeTests` ?
@tajila Can you review and merge this change ?

File: test/functional/Valhalla/testng.xml [hunk 1]
```diff
@@ -28,12 +28,20 @@
...
 			<class name="org.openj9.test.lworld.ValueTypeTests"/>
-			<class name="org.openj9.test.lworld.ValueTypeArrayTests"/>
 		</classes>
 	</test>
 	<test name="ValueTypeTestsJIT">
 		<classes>
 			<class name="org.openj9.test.lworld.ValueTypeTests"/>
+		</classes>
+	</test>
+	<test name="ValueTypeArrayTests">
```
Comment (llxia) [inline test/functional/Valhalla/testng.xml:38]: Do we need ValueTypeArrayTests and ValueTypeArrayTestsJIT? They have the same class - `org.openj9.test.lworld.ValueTypeArrayTests`

**Review by hzongaro — COMMENTED** [review]
Looks good.  I just wanted to suggest working with Annabelle to get the right set of variations in place for JIT testing.

File: test/functional/Valhalla/playlist.xml [hunk 2]
```diff
@@ -94,6 +94,79 @@
...
+			<disable>
+				<comment>https://github.com/eclipse-openj9/openj9/pull/9392#issuecomment-661246648</comment>
+				<variation>-Xjit:count=0</variation>
+			</disable>
+		</disables>
+		<variations>
+			<variation>-Xjit:count=0</variation>
+			<variation>-Xjit:count=1,disableAsyncCompilation -Xgcpolicy:optthruput</variation>
+			<variation>-Xjit:count=1,disableAsyncCompilation -Xgcpolicy:optthruput -XX:ValueTypeFlatteningThreshold=99999</variation>
+			<variation>-Xjit:count=1,disableAsyncCompilation -Xgcpolicy:gencon -XX:ValueTypeFlatteningThreshold=99999</variation>
```
Comment (hzongaro) [inline test/functional/Valhalla/playlist.xml:146]: Weird!  My review comment disappeared.

I just wanted to mention that Annabelle has pull request #16294 pending which adds variations for JIT testing.  You might want to coordinate these changes with her to ensure you have the right set of variations.

Comment (hangshao0) [reply test/functional/Valhalla/playlist.xml:146]: I will add the new JIT variations if #16294 is merged before this one.

Comment (a7ehuo) [reply test/functional/Valhalla/playlist.xml:146]: I think this PR can be merged before https://github.com/eclipse-openj9/openj9/pull/16294. I'll update https://github.com/eclipse-openj9/openj9/pull/16294 accordingly afterwards

**Review by hzongaro — APPROVED** [review]

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
Jenkins test sanity,extended xlinuxval jdknext

File: test/functional/Valhalla/testng.xml (hunk 1 above)
Comment (hangshao0) [reply test/functional/Valhalla/testng.xml:38]: > Yes, but it is the same class. Can we just reference one ValueTypeArrayTests in playlist? i.e., -testnames ValueTypeArrayTests

OK, I see what you mean. There is no need to have `ValueTypeArrayTestsJIT` (and `ValueTypeTestsJIT`) in this file. 
@tajila has just merged this PR. I will address this comment in another PR.

Comment (hangshao0) [reply test/functional/Valhalla/testng.xml:38]: > I will address this comment in another PR.

https://github.com/eclipse-openj9/openj9/pull/16307

**Pull request merged**